twake/backend/node/test/e2e/messages/messages.user-bookmarks.realtime.spec.ts

```diff
@@ -0,0 +1,137 @@
+      const roomToken = "twake";
+
+      connect();
+      socket.on("connect", () => {
+        socket
+          .emit("authenticate", { token: jwtToken })
+          .on("authenticated", () => {
+            socket.emit("realtime:join", {
+              name: `/companies/${platform.workspace.company_id}/messages/bookmarks`,
+              token: roomToken,
+            });
+            socket.on("realtime:join:error", () => done(new Error("Should not occur")));
+            socket.on("realtime:join:success", async () => {
+              await service.userBookmarks.save(
+                {
+                  company_id: platform.workspace.company_id,
+                  user_id: platform.currentUser.id,
+                  name: "mybookmarksaved",
+                  id: undefined,
+                },
+                {},
+                getContext(platform),
+              );
+            });
+            socket.on("realtime:resource", event => {
+              expect(event.type).toEqual("user_message_bookmark");
+              expect(event.action).toEqual("saved");
+              expect(event.resource.name).toEqual("mybookmarksaved");
+              done();
+            });
+          })
+          .on("unauthorized", () => {
+            done(new Error("Should not occur"));
+          });
+      });
+    });
+  });
+
+  describe("On bookmark removal", () => {
+    it("should notify the client", async done => {
+      const service = platform.platform.getProvider<MessageServiceAPI>("messages");
+
+      const instance = await service.userBookmarks.save(
+        {
+          company_id: platform.workspace.company_id,
+          user_id: platform.currentUser.id,
+          name: "mybookmark",
+          id: undefined,
+        },
+        {},
+        getContext(platform),
+      );
+
+      const jwtToken = await platform.auth.getJWTToken();
+      const roomToken = "twake";
+
+      connect();
+      socket.on("connect", () => {
+        socket
+          .emit("authenticate", { token: jwtToken })
+          .on("authenticated", () => {
+            socket.emit("realtime:join", {
+              name: `/companies/${platform.workspace.company_id}/messages/bookmarks`,
+              token: roomToken,
+            });
+            socket.on("realtime:join:error", () => done(new Error("Should not occur")));
+            socket.on("realtime:join:success", async () => {
+              await service.userBookmarks.delete(
+                {
+                  company_id: platform.workspace.company_id,
+                  user_id: platform.currentUser.id,
+                  id: instance.entity.id,
+                },
+                getContext(platform),
+              );
+            });
+            socket.on("realtime:resource", event => {
+              expect(event.type).toEqual("user_message_bookmark");
+              expect(event.action).toEqual("deleted");
+              done();
+            });
+          })
```

Comment:
This should be defined before the call to `userBookmarks.delete`

Reply:
Done, I moved the .emit("join") at the end so it does join -> realtime:join:success -> call service -> realtime:resource